### PR TITLE
Add previous flag check

### PR DESCRIPTION
**Pull request by ArtOfCode-**
Adds a check against a database of previously flagged posts. If a post is similar to a previously flagged post, it has a warning placed against it.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -17,20 +20,62 @@ $(document).ready(function() {
   	var autorefresh_time = 1000000000; //crazy long == no refresh, because I'm lazy
   	var autorefresh_timeout;
 	var sort = ByCreationDate;
+	var previousFlags;
+	var previousFlagText = "";
+	
+	refreshPreviousFlags = function(callback) {
+		$.ajax({
+			type: "POST",
+			url: "http://jenkinsstuff.bl.ee/blaze/get_data.php",
```
Comment (Undo1) [inline script.js]: Looks like you missed a hardcoded url - just `/blaze/get_data.php` ought to work.

File: submit_flagged.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	$post_text = mysqli_real_escape_string($conn, substr($_POST['post_text'], 0, 40000));
+	
+	if(!isset($_POST['post_text']) || $post_text == '') {
+		// #b.404/C: No post text provided to add to database.
+		http_response_code(400);
+		echo json_encode(array('error_code' => 'b.404+C', 'error_message' => 'No post text provided to add to database.'));
+		exit;
+	}
+	else {
+		$result = mysqli_query($conn, 'INSERT INTO Posts VALUES (DEFAULT, DEFAULT, \'' . $post_text . '\');');
```
Comment (Undo1) [inline submit_flagged.php]: Current table name is actually `posts` in MySQL, not sure if it's case sensitive or not.

**Comment by hichris1234** [comment]
I do think it would be a fair idea if @Undo1 were to host this -- not that I don't trust @ArtOfCode-, but that it prevents Blaze from being reliant on a third-party (i.e. if the other site were to go down, Blaze would lose functionality).

**Comment by ArtOfCode-** [comment]
@hichris1234 no problem with that, just needs some scripts and databases migrating to another host, and passwords and URL changes. Not hard.

**Pull request merged**